fractions/monitor/src/main/java/org/wildfly/swarm/monitor/runtime/HttpContexts.java

```diff
@@ -61,70 +99,171 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
         } else if (THREADS.equals(exchange.getRequestPath())) {
             threads(exchange);
             return;
-        } else if (Queries.isDirectAccessToHealthEndpoint(monitor, exchange.getRelativePath())) {
-            exchange.setStatusCode(StatusCodes.MOVED_PERMANENTLY);
-            exchange.getResponseHeaders().put(Headers.LOCATION, HEALTH + exchange.getRelativePath());
-            return;
         } else if (HEALTH.equals(exchange.getRequestPath())) {
-            listHealtSubresources(exchange);
+            proxyRequests(exchange);
             return;
-        } else if (exchange.getRelativePath().startsWith(HEALTH)) {
-            healthRedirect(exchange);
-            if(exchange.isResponseStarted()) // allow the redirect handler to proceed
-                return;
         }
 
         next.handleRequest(exchange);
     }
 
-    private void listHealtSubresources(HttpServerExchange exchange) {
+    private void proxyRequests(HttpServerExchange exchange) {
+
         if(monitor.getHealthURIs().isEmpty()) {
             noHealthEndpoints(exchange);
-        }
-        else
-        {
-            ModelNode payload = new ModelNode();
-            for (HealthMetaData endpoint : monitor.getHealthURIs()) {
-                payload.get("links").add(HEALTH + endpoint.getWebContext());
-            }
+        } else {
+
+            Runnable r = new Runnable() {
+                @Override
+                public void run() {
+
+                    try {
+
+                        final UndertowClient client = UndertowClient.getInstance();
+                        final List<ClientResponse> responses = new ArrayList();
+
+                        ClientConnection connection =
```

Comment:
Ah, so when a user hits `/health` or whatnot, we internally invoke additional HTTP requests back to ourself to aggregate the results?